checkEligibility.js

```diff
@@ -138,23 +145,55 @@ function sierraLogin () {
   })
 }
 
+/**
+ *  CheckEligibility
+ *  This is the primary method in this module.
+ *
+ *  @param {int} patronId The id patron for whom we wish to determine
+ *         hold-request eligibility.
+ *
+ *  @return {Promise<Hash>} A Promise that resolves a hash with a single
+ *          `eligible` property (boolean) indicating hold-request eligibility.
+ *
+ *  This method checks hold-request eligibility by performing two initial checks in parallel:
+ *   1. Can patron place a test hold on a fake item?
+ *   2. Does patron's ptype allow them to place holds?
+ */
 function checkEligibility (patronId) {
+  let patronInfo = null
+
   return config()
     .then(sierraLogin)
-    .then(() => initialCheck(patronId))
-    .then((eligible) => {
+    .then(() => Promise.all([
+      // Attempt a test hold:
+      patronCanPlaceTestHold(patronId),
+      // Simultaneously fetch patron info:
+      getPatronInfo(patronId)
+        .then((_patronInfo) => {
+          // Save patronInfo for later:
+          patronInfo = _patronInfo
+          // Determine whether/not ptype allowed to place holds:
+          return patronPtypeAllowsHolds(patronInfo)
+        })
+    ])).then((checks) => {
+      const [canPlaceTestHold, ptypeAllowsHolds] = checks
+      const eligible = canPlaceTestHold && ptypeAllowsHolds
```

Comment:
This looks good. Just to make sure I am following the logic, the issue here is that in this case, the patron can place a test hold even though the ptype doesn't allow holds, right?

Reply:
That's correct. We need to check *both* that a test hold succeeds and that the patron's ptype does not invalidate them for holds (because Sierra doesn't implement any restriction on the constrained ptypes)